### PR TITLE
Tickets/dm 30460

### PR DESCRIPTION
**Pull request by pcortes80**
Please review the code. Thanks.

### CODE REVIEW

**Review by teweitsai — CHANGES_REQUESTED** [review]
Please check the review txt in DM-30460. It looks like the behavior of simulator is changed totally. I did not see the received JSON command packet.

File: doc/versionHistory.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # Version History
 
+0.1.2
+
+- TCP Server sends noAck to TCP Client if command recived is not registered.
```
Comment (teweitsai) [inline doc/versionHistory.md:5]: recived  -> received

File: doc/versionHistory.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+- TCP Server replies noAck to TCP Client when cmdId and lastCmdId are different by more than 1 unit.
+- TCP Server sends ack to TCP Client for every commmad is received.
```
Comment (teweitsai) [inline doc/versionHistory.md:7]: TCP Server sends ack to TCP Client for every command is received if this command is registered.

**Review by teweitsai — APPROVED** [review]
Please fix the bugs we discussed before the merge for the dealing of the first cmdId and details of noAck message.

File: doc/versionHistory.md (hunk 1 above)
Comment (pcortes80) [reply doc/versionHistory.md:5]: Done.

File: doc/versionHistory.md (hunk 2 above)
Comment (pcortes80) [reply doc/versionHistory.md:7]: Done.

File: doc/versionHistory.md [hunk 3]
```diff
@@ -1,5 +1,12 @@
...
+- TCP Client decides which cmdId is the first, and TCP Server saves it as lastCmdId.
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/versionHistory.md:8]: Done.

**Pull request merged**